### DISCUSSION TITLE
Erro publishing app - status 500

### DISCUSSION THREAD

**Discussion by applicationuser** [Budibase/budibase#6459] [Help] [answered]
Error publishing app 

![image](https://user-images.githubusercontent.com/42922833/175696707-eb47d60c-683e-416f-825f-08224356ac30.png)

**Comment by melohagan** [comment]
Any further context?

**Reply by applicationuser** [reply]
Hello ,

Error while Publishing app to Production , receive error 'Error Publishing App' with spinner . 

network console has the 500 status error.  let me know if you need any further info.

App has budibase tables and few PostgreSQL DB queries

![image](https://user-images.githubusercontent.com/42922833/175994930-e9c3a76b-95f1-4b7b-ab89-c229429acfcd.png)

**Reply by melohagan** [reply]
Are you able to export the app, or is this the first time publishing this app?
Have you ever had issues publishing apps before? 
Is it only this app, or all your apps that you can't publish?
Are you self hosted or cloud?

If possible you could share your app export to support@budibase.com and I can take a closer look.

**Reply by applicationuser** [reply]
Are you able to export the app, or is this the first time publishing this app? **Can publish & export apps.  have another app that is published and active with no issues.**
Have you ever had issues publishing apps before? **No**
Is it only this app, or all your apps that you can't publish? **Only this app having issue.**
Are you self hosted or cloud? **Cloud**
app has data in table with export file size around 6MB app , would that be causing the issue?

is there any other way to share app ? app table data contains confidential information

**Reply by melohagan** [reply]
Are you using the Internal DB? In that case the data would be included in the app export so if you can't share don't worry about it. 

Can you think of anything at all that makes this app different to the other apps that can be published? 
I don't think the file size should be an issue at 6MB.

**Reply by applicationuser** [reply]
Yes using internal DB.  5 tables defined. one of the table has aprrox 900 records.
another difference from other published app is connection to postgres sql db to execute queries 
some of the pages use this db queries to pull data from postgres sql db for display ( do not store in budibase table)
automation's that run on user app action , Cron trigger

**Reply by melohagan** [reply]
There was another recent discussion stating issues with a Cron trigger. Could you try removing that automation if possible and seeing if that allows you to publish.
Otherwise maybe there is something unusual about the Postgres queries.

I don't think 900 records would do any harm.

**Reply by applicationuser** [reply]
Hello @melohagan 

automation with cron trigger might be possible issue, after deleting cron trigger based automations able to publish the app without any errors.

Thanks for your help

**Comment by melohagan** [comment, marked as answer]
Hi @applicationuser 

Thank you very much for helping to pinpoint the issue - I'll raise a bug. 

What version of Budibase are you using, and what version is your app?

**Reply by applicationuser** [reply]
App current version -  1.0.212 ( it was one version below when we had issue and did upgrade to this version before publishing app)

**Comment by 711pranjal** [comment]
Hi @applicationuser ,

Can you please check if you have any Cron jobs created under Automation?

If so, please see if you have "CRON TRIGGER" as blank field.

I was also facing this issue.
After doing RSA, I have figured out that I had "undefined" the trigger and so was giving undefined as value while doing the deployment.

@melohagan Request you to please see if we can add a "Disable" option in "CRON TRIGGER" field of automation, to overcome this unwanted behaviour of the BB.

<img width="956" alt="Cron job undefined" src="https://user-images.githubusercontent.com/41870509/179189210-cfe2e13d-6900-4b97-883f-9838cccad7e1.PNG">
@pueding @panos-- @opensas @Mitch-Budibase @shogunpurple @aptkingston @mjashanks @joebudi @jonnymccullagh @mike12345567 @kevmodrome @PClmnt

**Reply by melohagan** [reply]
Hi @pranjalgupta711 

Thanks very much for sharing, although if you wouldn't mind there's no need to tag every contributor 😀
